sound/soc/sof/intel/hda-codec.c

```diff
@@ -232,13 +232,16 @@ int hda_codec_i915_init(struct snd_sof_dev *sdev)
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	int ret;
 
-	/* i915 exposes a HDA codec for HDMI audio */
-	ret = snd_hdac_i915_init(bus);
-	if (ret < 0)
-		return ret;
-
-	/* codec_mask not yet known, power up for probe */
-	snd_hdac_display_power(bus, HDA_CODEC_IDX_CONTROLLER, true);
+	/* only do i915 codec init if it is available */
+	if (HDA_IDISP_CODEC(hda_codec_mask)) {
+		/* i915 exposes a HDA codec for HDMI audio */
+		ret = snd_hdac_i915_init(bus);
+		if (ret < 0)
+			return ret;
+
+		/* codec_mask not yet known, power up for probe */
+		snd_hdac_display_power(bus, HDA_CODEC_IDX_CONTROLLER, true);
```

Comment:
I don't get the logic here. We first test hda_codec_mask, but maybe it's not yet known still *after* we test?

Reply:
@plbossart maybe they mean here, that `.codec_mask` hasn't yet been set, but the module parameter is known of course

Reply:
@plbossart Yes, at this point the the `codec_mask` is not masked by `hda_codec_mask`
It is done in `hda_dsp_ctrl_init_chip`.
The over all sequence is like 
```
hda_dsp_probe
  hda_init
    hda_codec_i915_init
  hda_init_caps
    hda_dsp_ctrl_init_chip
```

https://github.com/thesofproject/linux/blob/4c37c24fb2976debdab1ffd4cfa2f1cf3db8d324/sound/soc/sof/intel/hda-ctrl.c#L239-L243

Reply:
I am not able to follow. Can you please update the comment or the logic to make sure there's no ambiguity here?

Reply:
@plbossart sure, I think the comment if from our old code, I will check if they are still useful here. 
From my understand, what we did here is to 
`/* power up i915 codec for probe and get codec_mask  */`